### PR TITLE
[Explore vis] add echarts implementation of metric chart

### PR DESCRIPTION
**Pull request by ruanyl**
### Description
Echarts implementation of metric visualization

## Changelog

- refactor: add echarts implementation of metric visualization

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

## Summary by CodeRabbit

* **New Features**
  * Metric charts gain an ECharts rendering option with threshold-driven colors, sparklines, automatic percentage-change displays, and improved unit-aware formatting.

* **Refactor**
  * Metric visualization layout updated to support compact metric cards, dynamic font sizing, and centralized rendering for consistent presentation.

* **Tests**
  * Extensive unit tests added for rendering, formatting, thresholds, sparklines, dynamic sizing, and edge cases.

* **Chores**
  * Added changelog fragment documenting the metric visualization changes.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/plugins/explore/public/components/visualizations/metric/to_expression.ts [hunk 1]
```diff
@@ -40,6 +43,32 @@ export const createSingleMetric = (
...
+      assembleSpec,
+      assembleForMetric
+    )({
+      data: transformedData,
+      styles,
+      axisConfig: { xAxis: dateColumn, yAxis: valueColumn },
+      axisColumnMappings: axisColumnMappings ?? {},
+    });
+    return result.spec;
+  }
```
[the remark this answers is not in the document]
Comment (ruanyl) [reply src/plugins/explore/public/components/visualizations/metric/to_expression.ts:70]: We will have separate task to track on how to improve visualization empty/invalid state

**Comment by wanglam** [comment]
Hi @ruanyl, it seems the unit tests for ciGroup4 failed. Could you help fix this? Thank you.

**Review by wanglam — APPROVED** [review]

File: src/plugins/explore/public/components/visualizations/metric/metric_utils.ts [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+  styles,
+}: {
+  seriesFields: string[];
+  styles: MetricChartStyle;
+  dateField?: string;
+}): PipelineFn => (state) => {
+  const { transformedData = [], axisColumnMappings } = state;
+  const newState = { ...state };
+
+  // If date field is set, it will a display sparkline
```
Comment (Hailong-am) [inline src/plugins/explore/public/components/visualizations/metric/metric_utils.ts:52]:
```suggestion
  // If date field is set, it will display a sparkline
```

File: src/plugins/explore/public/components/visualizations/metric/metric_utils.ts [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+
+    const selectedUnit = getUnitById(styles?.unitId);
+
+    const displayValue = showDisplayValue(isValidNumber, selectedUnit, calculatedValue);
+
+    const fillColor = targetFillColor(
+      styles?.useThresholdColor ?? false,
+      styles?.thresholdOptions?.thresholds,
+      styles?.thresholdOptions?.baseColor,
+      calculatedValue
```
Comment (Hailong-am) [inline src/plugins/explore/public/components/visualizations/metric/metric_utils.ts:86]: nit: styles is a required parameter, so styles?. is unnecessary. Use styles.useThresholdColor.

Comment (ruanyl) [reply src/plugins/explore/public/components/visualizations/metric/metric_utils.ts:86]: Thanks, I will tweak in next PR

**Comment by ruanyl** [comment]
> Vega implementation seems that have threshold related logic and seems it missing in echarts

Threshold logic is in `metric_utils.ts`, do you mean this?
